Haskell/Haskell.sublime-syntax

```diff
@@ -1,407 +1,1388 @@
+      scope:
+        meta.declaration.type.haskell
+        keyword.operator.big-arrow.haskell
+      set: type-signature
+    - include: type-expressions
+
+  type-signature:
+    - meta_include_prototype: false
+    - match: ''
+      set:
+        - type-args
+        - type-name
+
+  type-name:
+    - include: ident-namespaces
+    - match: '{{con_id}}'
+      scope: entity.name.type.haskell
+      captures:
+        1: storage.modifier.unboxed.haskell
+      pop: 1
+    - include: else-pop
+
+  type-args:
+    - meta_content_scope: meta.declaration.type.haskell
+    - match: (?={{operator_big_arrow}})
+      fail: type-declaration
+    - include: type-expressions
+    - include: else-pop
+
+###[ TYPE EXPRESSIONS ]########################################################
+
+  type-expressions:
+    - include: big-arrow-operators
+    - include: right-arrow-operators
+    - include: type-lists
+    - include: type-parens
+    - include: type-forall
+    - include: ident-anonymous
+    - include: ident-namespaces
+    - include: ident-types
+    - include: ident-variables
+
+  type-forall:
+    - match: forall{{break}}|∀(?!{{symbol}})
+      scope: keyword.control.forall.haskell
+      push: type-forall-body
+
+  type-forall-body:
+    # Note: This is the only situation a dot is allowed within type expressions.
+    - match: \.
+      scope: punctuation.separator.sequence.haskell
+      pop: 1
+    - include: ident-anonymous
+    - include: ident-variables
+    - include: else-pop
+
+  type-lists:
+    - match: \[
+      scope: punctuation.section.sequence.begin.haskell
+      push: type-list-body
+
+  type-list-body:
+    - meta_scope: meta.sequence.list.haskell
+    - include: list-end
+    - include: sequence-separators
+    - include: type-expressions
+    - include: else-pop
+
+  type-parens:
+    # 4.1.3 Syntax of Class Assertions and Contexts
+    # https://www.haskell.org/onlinereport/haskell2010/haskellch4.html#x10-630004.1
+    - include: empty-tuples
+    - include: prefix-parens-operators
+    - match: (?=\()
+      branch_point: type-parens
+      branch:
+        - type-group
+        - type-tuple
+
+  type-group:
+    - match: \(
+      scope: punctuation.section.group.begin.haskell
+      set: type-group-body
+
+  type-group-body:
+    - meta_scope: meta.group.haskell
+    - match: ',|:(?!{{symbol}})'
+      fail: type-parens
+    - include: group-end
+    - include: type-expressions
+    - include: else-pop
+
+  type-tuple:
+    - match: \(
+      scope: punctuation.section.sequence.begin.haskell
+      set: type-tuple-body
+
+  type-tuple-body:
+    - meta_scope: meta.sequence.tuple.haskell
+    - include: tuple-end
+    - include: sequence-separators
+    - include: type-expressions
+    - include: else-pop
+
+###[ BRACES ]##################################################################
+
+  blocks:
+    # 2.7 Layout
+    # https://www.haskell.org/onlinereport/haskell2010/haskellch2.html#x7-210002.7
+    - match: \{
```

Comment:
This also matches records, which I'm not sure is fine to be matched by it or not.

https://devtut.github.io/haskell/record-syntax.html

Reply:
Also here, the syntax can't distinguish those constructs by design.

Reply:
Can be resolved.